src/components/ebay-bar-chart/component.js

```diff
@@ -0,0 +1,346 @@
+                // return the proper label for each yAxis tick mark
+                formatter:
+                    this.input.yAxisLabels &&
+                    function () {
+                        if (this.isFirst) {
+                            yLabelsItterator = -1;
+                        }
+                        yLabelsItterator = yLabelsItterator + 1;
+                        return component.input.yAxisLabels[yLabelsItterator];
+                    },
+                style: {
+                    color: labelsColor, // setting label colors
+                },
+            },
+            maxVal,
+            title: {
+                enabled: false, // hide the axis label next to the axis
+            },
+            offset: 0, // set to zero for no offset refer to https://api.highcharts.com/highcharts/yAxis.offset
+            // passed in function for yAxisPositioner refer to https://api.highcharts.com/highcharts/yAxis.tickPositioner for use
+            tickPositioner: this.input.yAxisPositioner,
+        };
+    }
+
+    getLegendConfig() {
+        return {
+            symbolRadius: 2, // set corner radius of legend identifiers
+            enabled: this.input.series.length > 1, // disabled legend if only one series is passed in
+            itemStyle: {
+                color: legendColor, // set the color of the legend text
+            },
+            itemHiddenStyle: {
+                color: legendInactiveColor, // set the color of the legend text when clicked to hide
+            },
+            itemHoverStyle: {
+                color: legendHoverColor, // set the color of the legend text when hovering
+            },
+        };
+    }
+    tooltipFormatter() {
+        // function returns a function so it can keep a reference to the component scope
+        const component = this; // component reference used in formatter functions that don't have the same scope
+        return function () {
+            // refer to https://api.highcharts.com/class-reference/Highcharts.Time#dateFormat for dateFormat variables
+            // s is used to compile html string of formatted tooltip data
+            let s = `<b>${Highcharts.dateFormat('%b %e, %Y', this.x, false)}</b>`; // sets the displayed date at the top of the tooltip
+            if (component.input.stacked) {
+                // setup html for stacked tooltip
+                component.chartRef.series.forEach((serie) => {
+                    // cycle through each series
+                    serie.data.forEach((d) => {
+                        // cycle through each series data array to match x value with active hovered xAxis position
+                        if (d.x === this.x) {
+                            // when the x value matches the hovered xAxis position
+                            s += `<div style="display: flex; justify-content: space-between; width: 100%; align-items: flex-start;">${serie.name}<span style="margin-left: 16px">${d.label}</span></div>`;
```

Comment:
Is there a better way to do this?
I mean I really don't want to stick in a string of HTML in javascript. This seems very backwards to me.

Can we have a sub template which we can just import here and then render it to string?

Reply:
not with highcharts. It doesn't actually render html either.

for reference https://api.highcharts.com/highcharts/tooltip.formatter

quote from the docs 

"A subset of HTML is supported. Unless useHTML is true, the HTML of the tooltip is parsed and converted to SVG, therefore this isn't a complete HTML renderer. The following HTML tags are supported: b, br, em, i, span, strong. Spans can be styled with a style attribute, but only text-related CSS, that is shared with SVG, is handled."

Reply:
Marko has a renderToString. That should be better than this

```
Import subtemplate from 'subtemplate.marko'

subtemplate.renderToString(inputObject)
```

Should work